### PR TITLE
upgrade moto to v4.0.2

### PR DESCRIPTION
**Pull request by asjohnston-asf**
I haven't quite figured out *why* this particular configuration works, but it works. So far it only works if I include both the `@mock_dynamodb` decorator *and* the `with mock_dynamodb()` context manager.

See "pesky imports" at http://docs.getmoto.org/en/latest/docs/getting_started.html#what-about-those-pesky-imports

### CODE REVIEW

**Comment by asjohnston-asf** [comment]
@jtherrmann @jhkennedy I think this is a step forward, even if I'm not clear on the details? What are your thoughts on merging as-is?

File: requirements-all.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@
 -r requirements-apps-update-db.txt
 boto3==1.24.70
 jinja2==3.1.2
-moto==1.3.14
+moto[dynamodb]==4.0.2
```
Comment (jtherrmann) [inline requirements-all.txt:9]: Just out of curiosity what does the `[dynamodb]` do?

Comment (asjohnston-asf) [reply requirements-all.txt:9]: It filters which AWS services are installed, so limiting to just the `dynamodb` service we use to reduce the overall install footprint

http://docs.getmoto.org/en/latest/docs/getting_started.html
https://stackoverflow.com/questions/46775346/what-do-square-brackets-mean-in-pip-install

Comment (jtherrmann) [reply requirements-all.txt:9]: Nevermind, found it at http://docs.getmoto.org/en/latest/docs/getting_started.html

**Comment by jtherrmann** [comment]
https://github.com/ASFHyP3/hyp3/issues/1192

**Review by jtherrmann — APPROVED** [review]
I don't think there's a clear way to improve on this at the moment. See the issue linked above.

**Pull request merged**